packages/core/src/components/StepList/StepList.scss

```diff
@@ -0,0 +1,291 @@
+  }
+
+  &::after {
+    content: "";
+    display: block;
+    position: absolute;
+    bottom: 0;
+    left: -$step-list-padding-left;
+    right: -$step-list-padding-h;
+    height: 1px;
+    border-bottom: $step-border-width solid $step-border-color;
+  }
+}
+
+.ds-c-step--current,
+.ds-c-step--completed {
+  &::before {
+    line-height: $number-size;
+    border: none;
+    color: $color-white;
+  }
+}
+
+.ds-c-step--current {
+  .ds-c-step__header {
+    color: $color-primary;
+  }
+
+  &::before {
+    background: $color-primary;
+  }
+}
+
+.ds-c-step--completed {
+  color: $color-base;
+
+  .ds-c-step__header {
+    color: $color-base;
+  }
+
+  &::before {
+    background: $color-base;
+  }
+}
+
+.ds-c-step__header {
+  margin: 0;
+}
+
+.ds-c-step__title {
+  line-height: $step-button-height;
+}
+
+.ds-c-step__completed {
+  text-align: right;
+  font-size: $base-font-size;
+
+  span {
+    display: inline-block;
+    padding-top: 6px;
+    line-height: $step-button-height - 12px;
+  }
+
+  a {
+    display: block;
+    font-weight: $font-normal;
+  }
+
+  &::before {
+    background-size: contain;
+    background-image: url('#{$image-path}/checkmark-green.svg');
+    box-sizing: border-box;
+    content: '';
+    display: inline-block;
+    height: 12px;
+    margin-right: $spacer-1;
+    width: 16px;
+  }
+}
+
+.ds-c-step__description {
+  max-width: 75%;
+  font-size: $small-font-size;
+
+  .ds-c-step--current &,
+  .ds-c-step--completed & {
+    margin-top: $spacer-1;
+  }
+}
+
+.ds-c-step__substeps {
+  margin: 0;
+  padding-left: $spacer-5;
+  list-style-type: none;
+
+  .ds-c-step__description + & {
+    margin-top: $spacer-1;
+  }
+}
+
+.ds-c-substep {
+  display: block;
+  position: relative;
+  font-size: $small-font-size;
+
+  &::before {
+    content: "\2022";
+    position: absolute;
+    top: 0;
+    left: -$spacer-3;
+  }
+}
+
+/*
+Start, Resume, and Edit links
+
+A user interacts with the steps through "Start", "Resume", and "Edit"
+links.
+
+Visually "Start" and "Resume" are buttons, but all three are implemented
+with `<a>` tags. In the React component, the links' `href` properties are
+determined by the `step` object's `route` property. One can also
+optionally pass an `onEnterStep` function that will cancel the default
+link behavior and call `onEnterStep` with `route` as a parameter for
+apps that handle routing with JavaScript.
+
+When a step has substeps, the React component will automatically set the
+`href` property of the parent step's "Start" and "Resume" buttons to match
+the current (incomplete) substep's `route`. The "Edit" button will only
+appear on substeps that have been completed.
+
+Style guide: components.step-list.buttons
+*/
+
+/*
+`<StepList>`
+
+@react-component StepList
+
+Style guide: components.step-list.react
+*/
+
+/*
+Step object
+
+A `step` object contains all information needed to render the step, including
+text, progress, and routing information.
+
+Step object properties:
```

Comment:
For none of the sub-components though is the `step` object part of the root props object, so this won't make a difference.

Reply:
Oh hm, maybe I'm misunderstanding how this is all connected. I thought these were props of `Step.jsx`. I guess that would be the case if `step` wasn't a prop on `<Step>` and instead, the properties of the `step` object were top-level props on the `<Step>` component? Was there a reason for setting it up that way, rather than destructuring it like below?

```jsx
<Step onEnterStep={blah} resumeText="Resume" {...step} />
```

Reply:
I wanted to keep the `step` information all in its own object so it could be passed down to children in the hierarchy without getting muddied by the individual components' props—because not all components that take a step need the same props